RealmSwift-swift1.2/Tests/SwiftTestObjects.swift

```diff
@@ -60,35 +60,60 @@ class SwiftObject: Object {
 }
 
 class SwiftOptionalObject: Object {
-//    FIXME: Support all optional property types
-//    dynamic var optBoolCol: Bool?
-//    dynamic var optIntCol: Int?
-//    dynamic var optFloatCol: Float?
-//    dynamic var optDoubleCol: Double?
 #if REALM_ENABLE_NULL
-    dynamic var optStringCol: NSString?
+    dynamic var optNSStringCol: NSString?
+    dynamic var optStringCol: String?
     dynamic var optBinaryCol: NSData?
+    dynamic var optDateCol: NSDate?
+    let optIntCol = RealmOptional<Int>()
+    let optFloatCol = RealmOptional<Float>()
+    let optDoubleCol = RealmOptional<Double>()
+    let optBoolCol = RealmOptional<Bool>()
 #endif
-//    dynamic var optDateCol: NSDate?
     dynamic var optObjectCol: SwiftBoolObject?
 //    let arrayCol = List<SwiftBoolObject?>()
 }
 
 class SwiftImplicitlyUnwrappedOptionalObject: Object {
-//    FIXME: Support all optional property types
-//    dynamic var optBoolCol: Bool!
-//    dynamic var optIntCol: Int!
-//    dynamic var optFloatCol: Float!
-//    dynamic var optDoubleCol: Double!
 #if REALM_ENABLE_NULL
-    dynamic var optStringCol: NSString!
+    dynamic var optNSStringCol: NSString!
+    dynamic var optStringCol: String!
     dynamic var optBinaryCol: NSData!
+    dynamic var optDateCol: NSDate!
 #endif
-//    dynamic var optDateCol: NSDate!
     dynamic var optObjectCol: SwiftBoolObject!
 //    let arrayCol = List<SwiftBoolObject!>()
 }
 
+class SwiftOptionalDefaultValuesObject: Object {
+#if REALM_ENABLE_NULL
+    dynamic var optNSStringCol: NSString? = "A"
+    dynamic var optStringCol: String? = "B"
+    dynamic var optBinaryCol: NSData? = "C".dataUsingEncoding(NSUTF8StringEncoding)
+    dynamic var optDateCol: NSDate? = NSDate(timeIntervalSince1970: 10)
+    let optIntCol = RealmOptional<Int>()
+    let optFloatCol = RealmOptional<Float>()
+    let optDoubleCol = RealmOptional<Double>()
+    let optBoolCol = RealmOptional<Bool>()
+#endif
+    dynamic var optObjectCol: SwiftBoolObject? = SwiftBoolObject(value: [true])
```

Comment:
This doesn't actually work because during schema init `SwiftOptionalDefaultValuesObject` could be initialized before `SwiftBoolObject`, resulting in `initWithValue:` failing. Might be able to make it work, since during schema init the linked objects don't actually need to be any sort of sensible state.